src/store/sagas/dataset-list-saga.js

```diff
@@ -1,15 +1,15 @@
-import { takeEvery } from 'redux-saga/effects'
-import { RETRIEVE_DATA_LIST, selectDataList, displayError } from '../actions'
+import { takeEvery, select, take } from 'redux-saga/effects'
+import { DATASET_SEARCH, UPDATE_DATASET_SEARCH_PARAMS, datasetSearchSucceeded, displayError } from '../actions'
 import apiInvoker from './api-invoker'
+import { getSearchParams } from '../selectors'
+
 
 export default function * theRealDatasetSaga () {
-  yield takeEvery(RETRIEVE_DATA_LIST, apiInvoker({ endpoint: '/api/v1/dataset/search', actionator: selectDataList, queryParameterBuilder }))
+  while (true) {
+    yield take(UPDATE_DATASET_SEARCH_PARAMS)
```

Comment:
Is there a reason we have a separate action for this update? Do we not search every time we update the search params?